### ISSUE TITLE
[Feature] Settings backup

### ISSUE THREAD

**Issue by proydoha** [Youda008/DoomRunner#95]
When power outage happens there is a surprisingly high chance of losing an options.json file. Happened to me 4 times in last 6 months.

If possible it would be great if DoomRunner would automatically produce couple of backups that will be automatically loaded if main options.json is lost/unreadable/unparseable.

**Comment by Youda008** [comment]
Now this is a complete mystery to me.

The current file saving mechanism is designed to prevent exatly this case. How it works now is it
1. renames the existing file to `.old`,
2. creates a new file with the original name
3. writes the new state to the new file
4. only if the writing was successful, deletes the old file
All of this happens once per 60 seconds and on all modern computers and systems it shouldn't take more than few milliseconds.

In other words, the chance that hit that small millisecond window in a minute is extremely small and even if you do, there should always be a readable backup named `options.json.old`. If it doesn't, i have no idea what the hell is happening there.

Other programs also don't store backup files of their configs, there must be something else going in the background.
May i ask what's your operating system and drive?

**Comment by proydoha** [comment]
Windows 10, some Toshiba HDD. DT01ACA200 says device manager.

Can something go wrong in save/loading logic? For example maybe it reads malformed `options.json`, discards it and recreates `options.json` while ignoring `options.old.json`?

**Comment by Youda008** [comment]
Maybe, but you would get a warning message at the next start saying that the file is corrupted.
Anyway, when this happens next time, please look into the DoomRunner's folder before you start it, and send me the files that were there after the crash.
In the meantime, i will write code that will look for that `.old` backup and automatically load that instead, if the original file cannot be read.

**Comment by Youda008** [comment]
I release a version with yet another slightly reworked file saving mechanism. Please keep trying and report back if it happens again.

**Comment by Youda008** [comment]
I have reduced the frequency of writing to file, by making it write the content only when something has changed. Hopefully that will solve your issue.
